ibis/backends/base/__init__.py

```diff
@@ -136,17 +136,62 @@ class BaseBackend(abc.ABC):
     database_class = Database
     table_class: type[ops.DatabaseTable] = ops.DatabaseTable
 
+    def __init__(self, *args, **kwargs):
+        self._con_args: tuple[Any] = args
+        self._con_kwargs: dict[str, Any] = kwargs
+
+    def __getstate__(self):
+        return dict(
+            database_class=self.database_class,
+            table_class=self.table_class,
+            _con_args=self._con_args,
+            _con_kwargs=self._con_kwargs,
+        )
+
+    def __hash__(self):
+        return hash(self.datasource)
+
+    def __eq__(self, other):
+        return self.datasource == other.datasource
+
     @property
     @abc.abstractmethod
     def name(self) -> str:
         """
         Name of the backend, for example 'sqlite'.
         """
 
+    @property
+    def datasource(self) -> str:
```

Comment:
can we call this db_identity? (unless you have a definition for datasource here)

Reply:
@pcloud @gforsyth what do you think?  Is `db_identity` a better name than `datasource`?

Reply:
`db_identity` seems less semantically overloaded, and does track with the first line of the docstring

Reply:
Okay, changed and pushed.